custom/enikshay/two_b_release_1/tests.py

```diff
@@ -0,0 +1,247 @@
+from django.test import TestCase, override_settings
+
+from casexml.apps.case.const import CASE_INDEX_CHILD, CASE_INDEX_EXTENSION
+from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
+from corehq.apps.domain.models import Domain
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+from custom.enikshay.case_utils import get_parent_of_case, CASE_TYPE_DRTB_HIV_REFERRAL
+from custom.enikshay.const import ENROLLED_IN_PRIVATE
+from custom.enikshay.tests.utils import (
+    setup_enikshay_locations,
+    get_person_case_structure,
+    get_occurrence_case_structure,
+    get_episode_case_structure,
+    get_test_case_structure,
+    get_referral_case_structure,
+    get_trail_case_structure,
+)
+from .management.commands.enikshay_2b_case_properties import ENikshay2BMigrator
+
+
+@override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
+class TestCreateEnikshayCases(TestCase):
+    domain = 'enikshay-2b-migration'
+
+    def setUp(self):
+        self.project = Domain(name=self.domain)
+        self.project.save()
+        self.location_types, self.locations = setup_enikshay_locations(self.domain)
+        self.factory = CaseFactory(domain=self.domain)
+        self.cases = self.setup_cases()
+
+    def tearDown(self):
+        self.project.delete()
+
+    def setup_cases(self):
```

Comment:
Mainly just wanted to have more control over some of the properties set in the cases and how they related to each other.  I probably could've used it, but it felt like I would've ended up needing to override a lot of stuff anyways, and this felt more straightforward.